### ISSUE TITLE
Java Toolchain auto-provisioning fails with permission errors

### ISSUE THREAD

**Issue by jbartok** [gradle/gradle#22803]
```
Could not determine the dependencies of null.
> Could not resolve all dependencies for configuration ':classpath'.
  > Failed to calculate the value of task ':build-logic:lifecycle:compileJava' property 'javaCompiler'.
    > Unable to download toolchain matching the requirements ({languageVersion=11, vendor=ADOPTIUM, implementation=vendor-specific}) from 'https://api.adoptium.net/v3/binary/latest/11/ga/mac/aarch64/jdk/hotspot/normal/eclipse'.
      > Could not copy file '/Users/guy/.gradle/jdks/OpenJDK11U-jdk_aarch64_mac_hotspot_11/jdk-11.0.17+8/Contents/Home/legal/java.management.rmi/LICENSE' to '/Users/guy/.gradle/jdks/eclipse_adoptium-11-aarch64-os_x/jdk-11.0.17+8/Contents/Home/legal/java.management.rmi/LICENSE'.
        > /Users/guy/.gradle/jdks/eclipse_adoptium-11-aarch64-os_x/jdk-11.0.17+8/Contents/Home/legal/java.management.rmi/LICENSE (Permission denied)
```

Suspicion is that it happens when there is already content in the folder where we attempt to install it.

**Comment by davidburstrom** [comment]
@jbartok I've come across this issue as well, in a multiproject build where some projects use the same toolchain requirement. It looks to me like the toolchain engine doesn't propagate the fact that a JDK was just installed by one project, when another project is trying to query for it.

**Comment by davidburstrom** [comment]
This reproduces the issue 100%. [gradle-issue-22803-jdk-provisioning.zip](https://github.com/gradle/gradle/files/10531056/gradle-issue-22803-jdk-provisioning.zip)

Please see the README.md in the archive for how to run the build.

**Comment by davidburstrom** [comment]
I should add that it seems to affect Gradle 7.6. I just tested it with Gradle 8.0 RC 2 and it works as intended there.

**Comment by jbartok** [comment]
Should be fixed in the upcoming `7.6.1` version.

**Comment by davidburstrom** [comment]
Cool! If the reproducer doesn't fail in 7.6.1 I'd agree :)
